File: src/main/java/org/intellij/sdk/language/angelscript/AngelScriptElementFactory.java
package org.intellij.sdk.language.angelscript;

import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiFileFactory;
import org.intellij.sdk.language.angelscript.psi.AngelScriptFile;
import org.intellij.sdk.language.angelscript.psi.AngelScriptIdentifier;

public class AngelScriptElementFactory {
	public static AngelScriptIdentifier createIdentifier( Project project, String name ) {
		return (AngelScriptIdentifier) createFile( project, name ).getFirstChild();
	}

	public static AngelScriptFile createFile( Project project, String text ) {
		return (AngelScriptFile) PsiFileFactory.getInstance( project ).createFileFromText( "dummy.simple", AngelScriptFileType.INSTANCE, text );
	}
}

File: src/main/kotlin/org/stratasource/intellij/lang/angelscript/AngelScriptElementFactory.kt
package com.enderzombi102.css.lang.angelscript

import com.intellij.openapi.project.Project
import com.intellij.psi.PsiFileFactory
import com.enderzombi102.css.lang.angelscript.psi.AngelScriptFile
import com.enderzombi102.css.lang.angelscript.psi.AngelScriptIdentifier

object AngelScriptElementFactory {
	@JvmStatic
	fun createIdentifier(project: Project, name: String ) =
		createFile(project, name).firstChild as AngelScriptIdentifier

	private fun createFile(project: Project, text: String ) =
		PsiFileFactory.getInstance(project)
			.createFileFromText( "dummy.simple", AngelScriptFileType, text ) as AngelScriptFile
}